### PR TITLE
added open API documentation

### PR DESCRIPTION
**Pull request by raya311**
# Pull Request

## Summary

Added openAPI documentation. Integrated it with the Express server.

## Why

It's best practice to add OpenAPI documentation. It also provides access to others via Swagger UI.

## How to Verify

Provide clear steps to verify this change locally.

npm install
npm run server
http://localhost:3000/docs

## Checklist (Definition of Done)

- Branch follows team naming pattern (`week<NUM>/<short-description>`)
- PR opened early and kept reviewable in size
- At least one teammate review requested
- CI checks are passing
- Lint passes locally (`npm run eslint`)
- Tests pass locally (`npm run test`)
- New behavior includes starter tests or test plan notes
- Documentation updated if needed (`README`, `docs/`, ADRs)

## Notes for Reviewers

Anything specific you'd like feedback on?
- Edge cases?
- Architecture concerns?
- Naming?
- Test coverage?

### CODE REVIEW

File: docs/data/model.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- `name` (string): name of the trip.
+- `destinationType` (string): category of destination (e.g., "beach", "mountain").
+- `duration` (integer|null): number of days for the trip; `null` if unspecified.
+- `checklist` (array): list of checklist items; each item is an object with:
+	- `label` (string): text for the checklist entry.
+	- `completed` (boolean): whether the item is done.
+- `createdAt` (string): timestamp for when the trip was created.
+
+**Key fields**
+- destinationType, durantion and name are required fields.
```
Comment (hmhawn) [inline docs/data/model.md]: "durantion" misspelling and **Key Fields** is duplicated

File: package-lock.json
[the remark this answers is not in the document]
Comment (raya311) [reply package-lock.json]: we have dependency mentioned on ReadMe file with version V24LTS

**Comment by adari06** [comment]
Once Heather's comment is fixed, which is kind of a blocking issue, there are two more things that we should look at.

1. OpenAPI missing error responses
The spec only documents happy paths (200/201) plus the PUT 404. It's missing:

POST 400 (missing required fields)
POST 400 (invalid duration)
PUT 400 (invalid duration)
500 errors on all three endpoints
These are real responses the server returns and are tested. If we're documenting the API, the error cases are just as important for consumers.

2. No items schema on array fields
checklist is defined as type: array with no items schema, and GET /api/trips response is just type: array with no item definition. Swagger UI can't generate useful type info from this — it defeats the purpose of having a typed spec.

These should be easy to address after the first one is resolved.
